### PR TITLE
Fix Jetpack Free card styling issues

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR fixes a couple of styling issues in the Jetpack Free card. It also fixes the button link in Jetpack cloud.

Fixes 1169247016322522-as-1191767071741497

### Testing instructions

- Visit `/jetpack/connect/plans/:site/` with the offer reset flow enabled
- Check that the Jetpack Free card plan at the bottom of the page looks like the capture below, and that the button redirects to the WP admin
- Run Jetpack cloud and go to `/pricing/`
- Check that the Jetpack Free card plan at the bottom of the page looks like the capture below, and that the button redirects to the install page

#### In-place-connect

Before
Screen Shot 2020-09-02 at 11 13 18 AM
Screen Shot 2020-09-02 at 11 19 55 AM

After
Screen Shot 2020-09-02 at 12 02 18 PM

Screen Shot 2020-09-02 at 12 06 48 PM

#### Cloud

Before
Screen Shot 2020-09-02 at 11 24 55 AM
Screen Shot 2020-09-02 at 11 24 42 AM

After
Screen Shot 2020-09-02 at 12 03 03 PM

Screen Shot 2020-09-02 at 12 07 08 PM

### CODE REVIEW

File: client/components/jetpack/card/jetpack-free-card/style.scss [hunk 1]
```diff
@@ -29,24 +29,29 @@
 .jetpack-free-card__body {
 	padding: 24px 16px;
 
-	display: grid;
-	grid-template-rows: auto auto;
 	@include break-medium {
-		grid-template-columns: 50% 50%;
+		display: flex;
```
Comment (monsieur-z) [inline client/components/jetpack/card/jetpack-free-card/style.scss:33]: Grid isn't well supported by IE. I believe we still want to support the latest version.

File: client/components/jetpack/card/jetpack-free-card/style.scss [hunk 2]
```diff
@@ -29,24 +29,29 @@
...
 	}
+}
 
-	a {
-		text-align: center;
-		height: 40px;
-		@include break-medium {
-			margin-left: 24px;
-		}
+.jetpack-free-card__button {
```
Comment (monsieur-z) [inline client/components/jetpack/card/jetpack-free-card/style.scss:47]: The 'Start for free' button can be an `a` or `button`. It's better to use a class to style it.

File: client/components/jetpack/card/jetpack-free-card/index.tsx [hunk 3]
```diff
@@ -36,7 +37,12 @@ const JetpackFreeCard: FunctionComponent = () => {
 						}
 					) }
 				</p>
-				<Button href={ wpAdminUrl }>{ translate( 'Start for free' ) }</Button>
+				<Button
+					className="jetpack-free-card__button"
+					href={ wpAdminUrl || JPC_PATH_REMOTE_INSTALL }
```
Comment (monsieur-z) [inline client/components/jetpack/card/jetpack-free-card/index.tsx:42]: Defaults to the URL used at https://jetpack.com/pricing/.

File: packages/components/src/button/style.scss [hunk 4]
```diff
@@ -13,6 +13,7 @@
 	margin: 0;
 	outline: 0;
 	overflow: hidden;
+	text-align: center;
```
Comment (monsieur-z) [inline packages/components/src/button/style.scss:16]: This makes sure text is centered even when the tag is a `a` and not a `button`.

**Review by rcanepa — APPROVED** [review]
LGTM!

**Pull request merged**